core/src/models/Source.ts

```diff
@@ -428,13 +428,43 @@ export class Source extends LoggedModel<Source> {
   }
 
   @AfterSave
```

Comment:
I was thinking about removing the after save hook and just let this be triggered by the `afterSetMapping` hook instead. I don't think this needs to be updated every time the model is saved only when the mapping is set.

What do y'all think?